acceptance: TestDockerCLI_test_exec_log failed

acceptance.TestDockerCLI_test_exec_log [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_Acceptance/10819090?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_Acceptance/10819090?buildTab=artifacts#/) on master @ [1137aee0089fc2fe161bab0f1c899d2ddc997c90](https://github.com/cockroachdb/cockroach/commits/1137aee0089fc2fe161bab0f1c899d2ddc997c90):


```
=== RUN   TestDockerCLI_test_exec_log
    test_log_scope.go:167: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/130f713c8ffb7f103d68fcc03e847803/logTestDockerCLI_test_exec_log3924445248
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestDockerCLI_test_exec_log
    generated_cli_test.go:201: -- test log scope end --
--- FAIL: TestDockerCLI_test_exec_log (15.19s)
=== RUN   TestDockerCLI_test_exec_log/test_exec_log.tcl
=== CONT  TestDockerCLI_test_exec_log/test_exec_log.tcl
    util_docker.go:254: non-zero exit code: 1
    --- FAIL: TestDockerCLI_test_exec_log/test_exec_log.tcl (9.83s)
```
<p>Parameters: <code>GOFLAGS=-json</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_exec_log.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29533

--- a/pkg/cli/interactive_tests/common.tcl
+++ b/pkg/cli/interactive_tests/common.tcl
@@ -144,6 +144,20 @@ proc flush_server_logs {} {
     report "END FLUSH LOGS"
 }
 
+proc flush_and_sync_logs {filename grep_text} {
+    report "BEGIN FLUSH LOGS $filename"
+    system "kill -HUP `cat server_pid` 2>/dev/null"
+    # Wait for flush to occur.
+    system "for i in `seq 1 5`; do
+              grep '$grep_text' $filename && exit 0;
+              echo still waiting
+              sleep 1
+            done
+            echo 'server failed to flush logs?'
+            exit 1"
+    report "END FLUSH LOGS"
+}
+
 proc force_stop_server {argv} {
     report "BEGIN FORCE STOP SERVER"
     system "kill -KILL `cat server_pid`"

--- a/pkg/cli/interactive_tests/test_exec_log.tcl
+++ b/pkg/cli/interactive_tests/test_exec_log.tcl
@@ -97,23 +97,15 @@ send "SELECT 550+5;\r"
 eexpect 555
 eexpect root@
 
-flush_server_logs
+# Use this distinct query to be the boundary - if we see this stmt in the
+# exec log, then we should expect all the previous statements in the log too.
 
-# Now check the items are there in the log file. We need to iterate
-# because flush_server_logs only syncs on flush of cockroach.log, not
-# the exec log.
-#
-# We also check the last statement first, this ensures that every
-# previous statement is also in the log file after this check
-# succeeds.
-system "for i in `seq 1 3`; do
-  grep 'SELECT ..*550..* +' $logfile && exit 0;
-  echo still waiting;
-  sleep 1;
-done;
-echo 'not finding two separate txn counter values?';
-grep 'SELECT ..*550..* +' $logfile;
-exit 1;"
+send "SELECT 111;\r"
+eexpect 111
+eexpect root@
+
+flush_and_sync_logs $logfile "SELECT ..*111..*"
+
 
 # Two separate single-stmt txns.
 system "n=`grep 'SELECT ..*550..* +' $logfile | sed -e 's/.*TxnCounter.:\\(\[0-9\]*\\).*/\\1/g' | uniq | wc -l`; if test \$n -ne 2; then echo unexpected \$n; exit 1; fi"

